### PR TITLE
feat(pythonanywhere): init

### PR DESCRIPTION
**Pull request by somerand0mcat**
## 🎉 Theme for Pythonanywhere 🎉
a website that lets you run code and host python servers online
image

## 💬 Additional Comments 💬
im going to try to finish the port and add the blog along as some svgs
also going to make trinkey maintainer aswell (with permission from her)

## 🗒 Checklist 🗒

- [X] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [X] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [X] I have ensured that the new directory is in **lower-kebab-case**.
  - [X] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [X] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [X] I have included the following files:
  - [X] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [X] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

**Review by isabelroses — APPROVED** [review]
largly looks good. can't seem to notice any issues

File: scripts/userstyles.yml [hunk 1]
```diff
@@ -746,6 +746,14 @@ userstyles:
     readme:
       app-link: "https://pypi.org"
     current-maintainers: [*trinkey]
+  pythonanywhere:
+    name: PythonAnywhere
+    categories: [development]
+    color: sapphire
+    readme:
+      app-link: "https://pypi.org"
```
Comment (uncenter) [inline scripts/userstyles.yml:754]: Wrong link?

File: styles/pythonanywhere/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,1330 @@
...
+@author Catppuccin
+@license MIT
+
+@preprocessor less
+@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
+@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent" ["rosewater:Rosewater", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve*", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+==/UserStyle== */
+
+@import url("https://python.catppuccin.com/pygments/catppuccin-variables.important.css");
```
Comment (uncenter) [inline styles/pythonanywhere/catppuccin.user.css:18]: This doesn't go here.

File: styles/pythonanywhere/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,1330 @@
...
+      }
+      .ace_storage {
+        color: @peach;
+      }
+      .ace_string {
+        color: @green;
+      }
+      .ace_comment {
+        color: @surface2;
+      }
```
Comment (uncenter) [inline styles/pythonanywhere/catppuccin.user.css:170]: A bunch of these look wrong, make sure they match our other syntax highlighting ports.

File: styles/pythonanywhere/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,1330 @@
...
+  @macchiato: { @rosewater: #f4dbd6; @flamingo: #f0c6c6; @pink: #f5bde6; @mauve: #c6a0f6; @red: #ed8796; @maroon: #ee99a0; @peach: #f5a97f; @yellow: #eed49f; @green: #a6da95; @teal: #8bd5ca; @sky: #91d7e3; @sapphire: #7dc4e4; @blue: #8aadf4; @lavender: #b7bdf8; @text: #cad3f5; @subtext1: #b8c0e0; @subtext0: #a5adcb; @overlay2: #939ab7; @overlay1: #8087a2; @overlay0: #6e738d; @surface2: #5b6078; @surface1: #494d64; @surface0: #363a4f; @base: #24273a; @mantle: #1e2030; @crust: #181926; };
+  @mocha:     { @rosewater: #f5e0dc; @flamingo: #f2cdcd; @pink: #f5c2e7; @mauve: #cba6f7; @red: #f38ba8; @maroon: #eba0ac; @peach: #fab387; @yellow: #f9e2af; @green: #a6e3a1; @teal: #94e2d5; @sky: #89dceb; @sapphire: #74c7ec; @blue: #89b4fa; @lavender: #b4befe; @text: #cdd6f4; @subtext1: #bac2de; @subtext0: #a6adc8; @overlay2: #9399b2; @overlay1: #7f849c; @overlay0: #6c7086; @surface2: #585b70; @surface1: #45475a; @surface0: #313244; @base: #1e1e2e; @mantle: #181825; @crust: #11111b; };
+}
+// vim:ft=less
```
Comment (uncenter) [inline styles/pythonanywhere/catppuccin.user.css:1330]:
```suggestion
}

// vim:ft=less
```

File: styles/pythonanywhere/catppuccin.user.css [hunk 5]
```diff
@@ -0,0 +1,1330 @@
...
+    --ctp-subtext0: @subtext0;
+    --ctp-overlay2: @overlay2;
+    --ctp-overlay1: @overlay1;
+    --ctp-overlay0: @overlay0;
+    --ctp-surface2: @surface2;
+    --ctp-surface1: @surface1;
+    --ctp-surface0: @surface0;
+    --ctp-base: @base;
+    --ctp-mantle: @mantle;
+    --ctp-crust: @crust;
```
Comment (uncenter) [inline styles/pythonanywhere/catppuccin.user.css:1199]: Don't include these if there isn't code to be syntax highlighted, if there is code then add the import line.

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/pythonanywhere/catppuccin.user.css (hunk 5 above)
Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:1199]: there is code to be syntax highlighted

File: styles/pythonanywhere/catppuccin.user.css (hunk 2 above)
Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:18]: does this go to both the main website/blog?

File: styles/pythonanywhere/catppuccin.user.css (hunk 3 above)
Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:170]: this code appears to be the exact same as [this userstyle](https://github.com/catppuccin/userstyles/blob/195d5b63822209bf024cf91f655c60823ee0b5cc/styles/trinket/catppuccin.user.css#L224)

Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:170]: i checked one of the syntax highlighting ports and they appear to all match

Comment (uncenter) [reply styles/pythonanywhere/catppuccin.user.css:170]: Comments aren't supposed to be surface2.

File: styles/pythonanywhere/catppuccin.user.css (hunk 2 above)
Comment (uncenter) [reply styles/pythonanywhere/catppuccin.user.css:18]: Yes, as per the directions on https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments.

File: styles/pythonanywhere/catppuccin.user.css (hunk 5 above)
Comment (uncenter) [reply styles/pythonanywhere/catppuccin.user.css:1199]: > if there is code then add the import line.

File: styles/pythonanywhere/catppuccin.user.css (hunk 3 above)
Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:170]: then why does one of the syntax highlighting ports use ``surface2`` for its comments?
image

Comment (uncenter) [reply styles/pythonanywhere/catppuccin.user.css:170]: https://www.merriam-webster.com/dictionary/mistake

https://github.com/catppuccin/vscode/blob/2d7388ffe7321f588b2f54e2ffaab9fb7c9dde14/packages/catppuccin-vsc/src/theme/tokens/index.ts#L50-L57

Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:170]: > https://www.merriam-webster.com/dictionary/mistake
you don't need to link to the dictionary for me to understand its a mistake lol

File: styles/pythonanywhere/catppuccin.user.css [hunk 6]
```diff
@@ -0,0 +1,1333 @@
...
+      height: 0 !important;
+      padding-left: 41.45px !important;
+      padding-top: 30px !important;
+      background-image: url("data:image/svg+xml;utf8,@{svg}") !important;
+    }
+  }
+}
+
+@-moz-document domain('blog.pythonanywhere.com') {
+  @import url("https://python.catppuccin.com/pygments/catppuccin-variables.important.css");
```
[the remark this answers is not in the document]
Comment (somerand0mcat) [reply styles/pythonanywhere/catppuccin.user.css:1133]: i'll fix this after i've done the syntax highlighting

**Comment by trinkey** [comment]
2 things:
1. if you go to /user/[username]/files/[directory]?edit, the error text is unstyled (same for a permission denied file)
2. if you make a second account and add the first account as a teacher in the education tab in account settings, the switch account bar is unstyled

image

**Comment by somerand0mcat** [comment]
i cant fix the switch account bar because i dont have a education account on PA, can you give me the classname of that switch account bar?

**Comment by trinkey** [comment]
> can you give me the classname of that switch account bar?

```css
.teacher-navbar {
  background-color: @mantle !important;
  color: @text !important;
}
```

**Comment by somerand0mcat** [comment]
@trinkey fixed

**Comment by somerand0mcat** [comment]
ima close this pull request because of what happened recently and i dont wanna bring more chaos

**Pull request closed without merging**